MyCryptoHQ/MyCrypto
# Show Recent Txs on Check Tx Page

Pre-requisite to #1083, enhancement for #775.

### Description

When you send a transaction, it'll now be logged to local storage. Check Tx page uses this by displaying a select of your recent transactions.

### Changes

* Added `sagas/transaction.ts` function that listens to tx broadcasts, logs them to local storage
* Added select list to /tx-status, only visible when you have recent transactions
* Made /tx-status better at handling network changes
* Checksummed addresses shown on `components/TransactionStatus.tsx`

### Known Issues

* Right now the recent txs list is unlimited. We might want to put some kind of cap on it, like 100, and prune from it if transactions get old.
* The notification that pops up doesn't have a link to tx-status, I'll add that later.

### Steps to Test

1. Start app on the `save-transactions` branch
2. Send a transaction (Use testnet, works there too.)
3. Go to /tx-status
4. Confirm there is a dropdown
5. Select your TX from that dropdown
6. Confirm it was the correct transaction
7. Switch to a different network
8. Confirm that you have no selectable recent transactions

### Screenshots

![tx-history](https://user-images.githubusercontent.com/649992/36401149-7e04cb1e-15a3-11e8-8e4f-f0848d100097.gif)
